scheduler-utils/src/client/in-memory.js

```diff
@@ -25,33 +18,45 @@ export function createLruCache ({ size }) {
     sizeCalculation: (v) => JSON.stringify(v).length,
     allowStale: true
   })
-  return internalCache
+  return cache
 }
 
-export function getByProcessWith ({ cache = internalCache }) {
+/**
+ * @param {{ cache: LRUCache }} params
+ */
+export function getByProcessWith ({ cache }) {
   return async (process) => {
-    if (!internalSize) return
+    if (!cache.max) return
     return cache.get(process)
   }
 }
 
-export function setByProcessWith ({ cache = internalCache }) {
+/**
+ * @param {{ cache: LRUCache }} params
+ */
+export function setByProcessWith ({ cache }) {
   return async (process, { url, address }, ttl) => {
-    if (!internalSize) return
+    if (!cache.max) return
     return cache.set(process, { url, address }, { ttl })
   }
 }
 
-export function getByOwnerWith ({ cache = internalCache }) {
+/**
+ * @param {{ cache: LRUCache }} params
+ */
+export function getByOwnerWith ({ cache }) {
   return async (owner) => {
-    if (!internalSize) return
+    if (!cache.max) return
     return cache.get(owner)
   }
 }
 
-export function setByOwnerWith ({ cache = internalCache }) {
+/**
+ * @param {{ cache: LRUCache }} params
+ */
+export function setByOwnerWith ({ cache }) {
   return async (owner, url, ttl) => {
-    if (!internalSize) return
-    return cache.set(owner, { url, owner, ttl }, { ttl })
+    if (!cache.max) return
+    return cache.set(owner, { url, address: owner }, { ttl })
```

Comment:
I thought we were doing only a change to line 27 of locate

Reply:
Updating line 27 and also adding `ttl` to the `byOwner` cache entry. Otherwise, we will have no `ttl` on line 47.

https://github.com/permaweb/ao/pull/595#issuecomment-2047439206